src/dawn/CodeGen/GTClangCodeGen.cpp

```diff
@@ -81,6 +81,69 @@ GTClangCodeGen::IntervalDefinitions::IntervalDefinitions(const Stencil& stencil)
                               Interval::makeCodeGenName(stencilFun->getInterval()));
 }
 
+class StencilFunctionReader : public ASTCodeGenGTClangStencilBody {
+private:
+  std::shared_ptr<sir::StencilFunction> function;
+
+public:
+  using Base = ASTCodeGenGTClangStencilBody;
+  StencilFunctionReader(const StencilInstantiation* stencilInstantiation,
```

Comment:
Is this related to BC only? Could we have a more descriptive name ?  Also Reader sounds strages when it does `ss << `

Reply:
It is somewhat unique to BC's since it specifically parses the stencil function and rewrites the field-accesses to be the templated parameters. This is not needed anywhere else

As for the name, is StencilFunctionToBoundaryConditionConverter useful? is it too long? What would you suggest?